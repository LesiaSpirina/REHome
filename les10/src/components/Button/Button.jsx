import  PropTypes  from 'prop-types'
import styles from './Button.module.css'

export function Button(props) {
    
    return (
        <>
        <button  {...props}  style={{backgroundColor: 'rgb(163, 102, 163)'}} className={styles.button}
      >{props.children}</button>
        </>
    )
}

Button.propTypes = {
    type: PropTypes.string
}