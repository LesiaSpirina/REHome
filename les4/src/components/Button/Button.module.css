.button{
    
    background-color: rgb(163, 102, 163);
    color: rgb(237, 225, 236);
    padding: 10px;
    border: 1px solid rgb(50, 6, 45);
}