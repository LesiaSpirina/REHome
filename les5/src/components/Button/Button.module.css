.button{
    
    background-color: rgb(163, 102, 163);
    color: rgb(237, 225, 236);
    padding: 10px;
    border: 0;
    border-radius: 4px
}