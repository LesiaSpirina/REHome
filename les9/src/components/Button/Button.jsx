import  PropTypes  from 'prop-types'
import styles from './Button.module.css'

export function Button(props) {
    return (
        <>
        <button  {...props} className={styles.button}
         onClick={props.click}>{props.children}</button>
        </>
    )
}

Button.propTypes = {
    type: PropTypes.string
}