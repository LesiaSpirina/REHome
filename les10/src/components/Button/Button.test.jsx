import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import { Button } from "./Button";

describe('Testing Button', () => {
    it('render component Button', () => {
        render(<Button>Change Theme</Button>)
    })
    it('render with snapshot', () => {
        const {asFragment} = render(<Button>Change Theme</Button>)
        expect(asFragment()).toMatchSnapshot()
    })
    it('render component wiht text name button', () => {
        render(<Button>Change Theme</Button>)
        expect(screen.getByText(/Change Theme/)).toBeInTheDocument()
    })
    it('render multiply component', () => {
        render (
            <> 
            <Button>Change Theme</Button>
            <Button>Add message</Button>
            </>
        )
        expect(screen.queryAllByRole('button').length).toBe(2)
    })
    it('button is disable', () => {

        render(<Button disabled>Delete</Button>)
        expect(screen.getByText(/Delete/)).toBeDisabled()
    })
    it('button have style backgroundColor rgb(163, 102, 163)', () => {
        render(<Button>Add message</Button>)
        expect(screen.getByText(/Add message/)).toHaveStyle({ backgroundColor: 'rgb(163, 102, 163)'})
    })
    // it('button click with userEvent', async () => {
    //     const mockHandler = jest.fn()
    
    //     render(<Button click={mockHandler}>Delete</Button>)
    //     await userEvent.click(screen.getByText(/Delete/))
    //     expect(mockHandler).toHaveBeenCalledTimes(1)
    //   })
    it('test checkbox on double click', async() => {
        const onChange = jest.fn()
        render(<input type="checkbox" onChange={onChange} />)

        const checkbox = screen.getByRole('checkbox')
        await userEvent.dblClick(checkbox)
        expect(onChange).toHaveBeenCalledTimes(2)
        expect(checkbox).not.toBeChecked()
    })
})