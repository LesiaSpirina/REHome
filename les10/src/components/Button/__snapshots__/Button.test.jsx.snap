// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    style="background-color: rgb(163, 102, 163);"
  >
    Change Theme
  </button>
</DocumentFragment>
`;
